plugins/BulkTracking/Tracker/Handler.php

```diff
@@ -96,4 +104,23 @@ private function isTransactionSupported()
         return (bool) TrackerConfig::getConfigValue('bulk_requests_use_transaction');
     }
 
+    private function isBulkTrackingRequestAuthenticated(RequestSet $requestSet)
+    {
+        $tokenAuth = $requestSet->getTokenAuth();
+        if (empty($tokenAuth)) {
+            return false;
+        }
+
+        Piwik::postEvent('Request.initAuthenticationObject');
+
+        /** @var \Piwik\Auth $auth */
+        $auth = StaticContainer::get('Piwik\Auth');
+        $auth->setTokenAuth($tokenAuth);
+        $auth->setLogin(null);
+        $auth->setPassword(null);
+        $auth->setPasswordHash(null);
+        $access = $auth->authenticate();
+
+        return $access->getCode() != AuthResult::FAILURE;
```

Comment:
Sorry I didn't understand the code, I realized this is used in `Response` and just for the output therefore view permission might be okay. In theory we should then return only details about invalid requests where this token has view access to (meaning only for some sites) unless it's superuser? 

Maybe it's not needed to restrict it that much but would maybe at least exclude `anonymous` token? Not sure... Imagine someone is running a Piwik with many different, independent users (like a Piwik Cloud) in one big database... they could possibly see each other's invalid reports because each one has view access to that instance? Some users are using Piwik this way but I'm really not sure re implications here